client/utils/tokens.ts

```diff
@@ -0,0 +1,17 @@
+import type { Tokens } from '~/stores/useAuth'
+
+export function getTokensFromLocalStorage(): Tokens | null {
+  const auth = localStorage.getItem('auth')
+  if (!auth)
+    return null
+
+  const parsed = JSON.parse(auth)
+  if (!parsed.tokens || !parsed.tokens.accessToken || !parsed.tokens.refreshToken)
+    return null
+
+  return parsed.tokens
+}
```

Comment:
cuz of a circular dependency

- useAuth needs useApi
- useApi needs useAxiosInstance
- useAxiosInstance needs useAxiosMiddlewares
- useAxiosMiddlewares needs the tokens, but can't use useAuth since it depends on it

it is what it is

Reply:
Yes, but I mean you could use Pinia directly to read the 'auth' store, and thus avoid issues if the store storage method changes in the future..

Reply:
useAuth is the pinia store which manages that

and as I said, if useAxiosMiddlewares were to use it, it would cause a circular dependency.

The coupling is not that big anyways to be fair. If it does become anoying at some point, I know of a way to solve it, which ill do later. (making a custom storage serializer for pinia persistance)

Reply:
if you meant to extract the logic of the storage only, it then completely defeats the purpose of the store to encapsulate it

Reply:
Oh yeah I forgot `useAuth` was in fact a Pinia store.. I though for a moment it was some sort of a wrapper, so you're right.